apigee_api_catalog.module

```diff
@@ -40,3 +43,32 @@ function apigee_api_catalog_help($route_name, RouteMatchInterface $route_match)
     default:
   }
 }
+
+/**
+ * Implements hook_jsonapi_ENTITY_TYPE_filter_access() for 'apidoc'.
+ */
+function apigee_api_catalog_jsonapi_apidoc_filter_access(EntityTypeInterface $entity_type, AccountInterface $account) {
+  // Allow admins to see all entities.
+  if ($admin_permission = $entity_type->getAdminPermission()) {
+    if ($account->hasPermission($admin_permission)) {
+      return ([
+        JSONAPI_FILTER_AMONG_ALL => AccessResult::allowedIfHasPermission($account, $admin_permission),
+      ]);
+    }
+  }
+
+  // Allow all entities if account has view published and unpublished or
+  // account had API Catalog admin permission.
+  if ($account->hasPermission('view published apidoc entities') &&
+    $account->hasPermission('view unpublished apidoc entities')) {
+    return([
+      JSONAPI_FILTER_AMONG_ALL => AccessResult::allowed(),
```

Comment:
Add the the "user.permissions" cache context: `AccessResult::allowed()->cachePerPermissions()`